Microsoft.Toolkit/Extensions/StringExtensions.cs

```diff
@@ -19,6 +19,10 @@ namespace Microsoft.Toolkit.Extensions
     /// </summary>
     public static class StringExtensions
     {
+        internal const string DecimalRegex = "^-?[0-9]{1,28}([.,][0-9]{1,28})?$";
+        internal const string NumberRegex = "^-?[0-9]{1,9}$";
```

Comment:
@nmetulev I just noticed these have length limits on them, was there a reason for that?  Do you know?

Reply:
probably to limit to int

Reply:
But int32 can be 10 digits...

Would it be just better if we used `int.TryParse` and `double.TryParse` for these two functions?

Reply:
i think decimals have 29 digits of precision. Probably that's why?

Also According to [this](https://stackoverflow.com/a/969400/4585476) answer on SO, `decimal.TryParse` is faster than Regular expression. So we should consider changing this to Parsing than Regular Expressions.